### PR TITLE
Update Developer Guide delete command

### PR DESCRIPTION
**Pull request by ElginL**

### CODE REVIEW

**Review by notle1706 — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -288,6 +288,50 @@ The following activity diagram summarizes what happens when a user executes the
     * Cons: User would have to manually select each client and filter the transactions.
 
 _{more aspects and alternatives to be added}_
+
```
Comment (quannam0124) [inline docs/DeveloperGuide.md:365]: Great explanation and clear diagram! LGTM Elgin!

**Review by quannam0124 — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -288,6 +288,50 @@ The following activity diagram summarizes what happens when a user executes the
...
+### Delete Client/Transaction/Remark feature
+
+#### Current Implementation
+
+The deletion mechanism for `clients`, `transactions`, and `remarks` is facilitated by a `DeleteCommandParser` and `DeleteCommand`.
+
+The following class diagram shows the parent-child relation of `DeleteClientCommand`, `DeleteTransactionCommand`, `DeleteRemarkCommand` relative to the `DeleteCommand`. The concrete classes consists of the logic to delete an item stated by the name of their command.
+
```
Comment (Jo3LW) [inline docs/DeveloperGuide.md:373]: Great implementation of the DG!

**Review by Jo3LW — APPROVED** [review]
LGTM! :)

**Pull request merged**